colour/colorimetry/spectrum.py

```diff
@@ -1939,8 +2050,49 @@ def align(self,
                 77.18     ...,  77.18     ...,  77.18     ...,  77.18     ...])
         """
 
-        self.interpolate(shape)
-        self.extrapolate(shape, method, left, right)
+        self.interpolate(shape, interpolation_method)
+        self.extrapolate(shape,
+                         extrapolation_method,
+                         extrapolation_left,
+                         extrapolation_right)
+
+        return self
+
+    def trim(self, shape):
```

Comment:
@Ron024: I considered `clip` but it is used for array values in Numpy, I would expect it to affect the *values* not the *wavelengths* because of that.

I quite like `trim` and `cull` because I usually associate those actions as acting on the length of an entity, making it shorter.

But we are probably better off being explicit than implicit. Thinking about the future and with the [spectral signal code](https://github.com/colour-science/colour-ramblings/blob/master/spectral_signal.ipynb) in mind.

I would suggest to name it `trim_domain` or `trim_wavelengths` which would be a convenient alias to `Signal.trim_domain` method. We can take that opportunity to introduce explicit methods like `align_wavelengths` for example.